### ISSUE TITLE
Translation framework

### ISSUE THREAD

**Issue by Lyonlancer5** [Lenidevelaire/opsu-variant#2]
# Overview

> the world isn't English-only, _riiiiiiight_

Add a way to support translating the game itself, including notifications, the UI and certain game elements. This way, we could let users fully understand the parts and things of the game.

# Stuff to do
- [ ] Fully integrate this framework into the main codebase
- [ ] ~~Allow support for "automatic updating" of translation files~~
- [ ] Let users create their own translation files
- [ ] Expose the framework to be used in other gameplay-related elements

**Comment by Lyonlancer5** [comment]
https://github.com/itdelatrisu/opsu/pull/295 - see this pull request for more information

Close this issue once system is in place.

**Comment by Lyonlancer5** [comment]
Marking off point 3 because of #5

**Comment by Lyonlancer5** [comment]
Redoing because original PR went out-of-scope
